vita3k/util/src/net_utils.cpp

```diff
@@ -472,4 +478,73 @@ bool download_file(const std::string &url, const std::string &output_file_path,
     return res == CURLE_OK;
 }
 
+void getAllAssignedAddrs(std::vector<AssignedAddr> &outAddrs) {
+    outAddrs.clear();
+#ifdef _WIN32
+    PIP_ADAPTER_INFO pAdapterInfo;
+    DWORD dwRetVal = 0;
+    UINT i;
+    ULONG ulOutBufLen = sizeof(IP_ADAPTER_INFO);
+    pAdapterInfo = (IP_ADAPTER_INFO *)malloc(sizeof(IP_ADAPTER_INFO));
+    if (pAdapterInfo == NULL) {
+        LOG_CRITICAL("Error allocating memory needed to call GetAdaptersinfo");
+        return;
+    }
+    // Make an initial call to GetAdaptersInfo to get the necessary size into the ulOutBufLen variable
+    if (GetAdaptersInfo(pAdapterInfo, &ulOutBufLen) == ERROR_BUFFER_OVERFLOW) {
+        free(pAdapterInfo);
+        pAdapterInfo = (IP_ADAPTER_INFO *)malloc(ulOutBufLen);
+        if (pAdapterInfo == NULL) {
+            LOG_CRITICAL("Error allocating memory needed to call GetAdaptersinfo");
+            return;
+        }
+    }
+    if ((dwRetVal = GetAdaptersInfo(pAdapterInfo, &ulOutBufLen)) == NO_ERROR) {
+        PIP_ADAPTER_INFO pAdapter = pAdapterInfo;
+        const std::string noAddress = "0.0.0.0";
+        while (pAdapter) {
+            IP_ADDR_STRING *pIPAddr = &pAdapter->IpAddressList;
+            while (pIPAddr) {
+                if (noAddress.compare(pIPAddr->IpAddress.String) != 0)
+                    outAddrs.push_back({ pAdapter->Description, pIPAddr->IpAddress.String, pIPAddr->IpMask.String });
+                pIPAddr = pIPAddr->Next;
+            }
+            pAdapter = pAdapter->Next;
+        }
+    } else {
+        LOG_CRITICAL("GetAdaptersInfo failed with error: %d", dwRetVal);
+    }
+    if (outAddrs.size() == 0)
```

Comment:
Put in in end like that, no need duplicate same for linxu and windows

if (outAddrs.empty())
        outAddrs.push_back({ "localhost", "127.0.0.1", "255.255.255.255" });